lib/ngx/re.lua

```diff
@@ -153,6 +160,9 @@ function _M.split(subj, regex, opts, ctx, max, res)
     local res_idx = 0
     local last_empty_match
 
+    -- update to split_helper indexes
+    ctx.pos = ctx.pos - 1
```

Comment:
Should we use `sub_idx` here on the RHS to avoid a duplicate `ctx.pos` table lookup?

Reply:
Fine with me, updated.